pandas/core/algorithms.py

```diff
@@ -541,7 +547,11 @@ def factorize_array(
 
     table = hash_klass(size_hint or len(values))
     uniques, codes = table.factorize(
-        values, na_sentinel=na_sentinel, na_value=na_value, mask=mask
+        values,
+        na_sentinel=na_sentinel,
+        na_value=na_value,
```

Comment:
for the list of things to improve signature-clarity-wise, na_sentinel vs na_value is confusing

Reply:
Is it just the names or descriptions in the docstring? If it's the name, perhaps na_code instead of na_sentinel. Or are you thinking more we shouldn't have both as arguments?

As I was working on this, I was wondering what use case na_sentinel has. In other words, what's the downside of having na_sentile always be -1 and hence not an argument.